dvc/repo/ls.py

```diff
@@ -0,0 +1,107 @@
+import os
+
+from dvc.exceptions import DvcException, NotDvcRepoError, OutputNotFoundError
+from dvc.scm.base import CloneError
+
+
+@staticmethod
+def ls(url, target=None, rev=None, recursive=None, outputs_only=False):
+    if os.path.exists(url):
+        if rev:
+            raise DvcException("Local repo doesn't support revisions")
+        return _ls_local_dir(url, target, recursive, outputs_only)
+
+    try:
+        return _ls_remote_git_repo(url, target, rev, recursive, outputs_only)
+    except CloneError:
+        raise DvcException("There is no such url {}".format(url))
+
+
+def _ls_files(path, target=None, recursive=None):
+    from dvc.scm.tree import WorkingTree
+    from dvc.ignore import CleanTree
+
+    def prettify(prefix, filename):
+        if not prefix == os.curdir:
+            filename = os.path.join(prefix, filename)
+        return filename
+
+    files = []
+    target_path = os.path.join(path, target) if target else path
+    tree = CleanTree(WorkingTree(target_path))
+
+    try:
+        target_path = os.path.join(path, target) if target else path
+        if not os.path.exists(target_path):
+            raise FileNotFoundError("{} doesn't exist".format(target_path))
+
+        if target:
+            files.append(prettify(os.curdir, target))
+            if not recursive:
+                return files
+
+        for dirpath, dirnames, filenames in tree.walk(target_path):
+            prefix = os.path.relpath(dirpath, path)
+            files.extend(map(lambda f: prettify(prefix, f), dirnames))
+            files.extend(map(lambda f: prettify(prefix, f), filenames))
+            if not recursive:
+                break
+    except (NotADirectoryError, FileNotFoundError):
+        pass
+
+    return files
+
+
+def _ls_outs_local_repo(repo, target=None, recursive=None):
+    def prettify(out):
```

Comment:
Naming is misleading here. It doesn't prettify but rather extracts first part of the path in non-recursive case.

Reply:
Not only, it outs the list of strings for the pointed output
In case of non-recursive call the response contains the topest dir and in recursive - list of segments to the out. So it takes out and returs it's representation list
I've named it such way cause we have similar function in `_ls_files`, and purpose of it was prettify out somehow. I think it can be renamed to `get_segments`